### ISSUE TITLE
log base can be expressed more clearly

### ISSUE THREAD

**Issue by shionryuu** [rust-lang/rust-clippy#11728]
### Description

```rust
pub fn calc(x: i32, y: i32) -> i32 {
    let y = f64::from(y);
    let x = f64::from(x);
    (y.log2() / x.log2()).ceil() as i32
}

pub fn calc2(x: i32, y: i32) -> i32 {
    let y = f64::from(y);
    let x = f64::from(x);
    y.log(x).ceil() as i32
}

fn main() {
    println!("{} {}", calc(5, 125), calc2(5, 125)); // 3 4
}
```

run the above code with cippy will give `clippy::suboptimal-flops` error, but the expression suggest by clippy is not equivalent to original one

```text
error: log base can be expressed more clearly
   |
 4 |     (y.log2() / x.log2()).ceil() as i32
   |     ^^^^^^^^^^^^^^^^^^^^^ help: consider using: `y.log(x)`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#suboptimal_flops
   = note: `-D clippy::suboptimal-flops` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::suboptimal_flops)]`
```

### Version

```text
rustc 1.73.0 (cc66ad468 2023-10-03)
binary: rustc
commit-hash: cc66ad468955717ab92600c770da8c1601a4ff33
commit-date: 2023-10-03
host: x86_64-unknown-linux-gnu
release: 1.73.0
LLVM version: 17.0.2
```

**Comment by asquared31415** [comment]
`f64::log` notes that rounding may be incorrect (different than the float that you would get if all operations were infinite precision).  This is probably true of *all* floating point operations, perhaps `suboptimal_flops` should add a note that the results may be slightly different in any case that you change the exact operations done?
